include/platform.hpp

```diff
@@ -56,4 +56,9 @@
 	#define setmode(fd, mode) (0)
 #endif
 
+// MingGW and Cygwin need POSIX functions which are not standard C explicitly enabled
+#if defined(__MINGW32__) || defined(__MINGW32__) || defined(__CYGWIN__)
```

Comment:
@Rangi42 Why are you checking `__MINGW32__` twice?

Reply:
In case the first one fails. :3

Reply:
Fixed in c7a029a0515b87725926925b6d45af1a619d5f5d.